applications/nrf_desktop/src/hw_interface/click.c

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+
+#define MODULE click
+#include "module_state_event.h"
+#include "button_event.h"
+#include "click_event.h"
+
+#include "click_def.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_CLICK_LOG_LEVEL);
+
+#define SHORT_CLICK_MAX		K_MSEC(500)
+#define LONG_CLICK_MIN		K_MSEC(5000)
+#define CLICK_ON_INIT_MAX	K_MSEC(500)
+#define HOLD_ON_INIT_MIN	K_MSEC(5000)
+
+enum state {
+	STATE_DISABLED,
+	STATE_INIT_CLICK_CHECK,
+	STATE_INIT_HOLD_CHECK,
+	STATE_ACTIVE,
+};
+
+struct click_detector {
+	const struct click_detector_config *config;
+	struct k_delayed_work single_click;
```

Comment:
please use single work and timestamp

Reply:
switch to single work and structure remembering state of every button